File: tools/tarkov-runtime-mcp/src/logs/log-entry.ts
// =============================================================================
// 日志条目（MCP 侧三通道统一形状）
//
// 服务器日志 tail 与 fatal 通道解析后都产出本形状；`ts` 一律为 UTC ISO 8601
// （服务器日志由行内本地时间转换，fatal 通道取观测时刻），以便与桥侧 `ts`
// 在同一视图内比较、排序与按 `since` 过滤。
// =============================================================================

/** 已采集日志条目（字段序稳定：ts/level/source/text） */
export interface LogEntry {
  /** 观测时刻（UTC ISO 8601） */
  ts: string;
  /** 归一化小写级别名（fatal/error/warning/message/info/debug） */
  level: string;
  /** 来源标识（服务器日志为 `server:<文件名>`，fatal 通道为 `fatal`） */
  source: string;
  /** 原始文本（多行条目以 \n 拼接） */
  text: string;
}

File: tools/tarkov-runtime-bridge/src/LogEntry.cs
using System;

namespace SamMeow.TarkovRuntimeBridge;

/// <summary>
/// 单条捕获的日志（不可变）。<see cref="Seq"/> 由 <see cref="LogRingBuffer"/> 单调分配，
/// <see cref="TimestampUtcTicks"/> 为捕获时的墙钟 UTC Ticks，
/// <see cref="Level"/> 为 <see cref="LogWatchLevel"/> 归一化后的小写级别名（JSON 稳定字面量）。
/// </summary>
internal sealed class LogEntry
{
    internal LogEntry(long seq, long timestampUtcTicks, string level, string source, string text)
    {
        Seq = seq;
        TimestampUtcTicks = timestampUtcTicks;
        Level = level ?? string.Empty;
        Source = source ?? string.Empty;
        Text = text ?? string.Empty;
    }

    internal long Seq { get; }

    internal long TimestampUtcTicks { get; }

    /// <summary>归一化级别名：fatal / error / warning / message / info / debug（小写）。</summary>
    internal string Level { get; }

    /// <summary>日志来源（BepInEx <c>ILogSource.SourceName</c>，如 Unity / Assembly-CSharp）。</summary>
    internal string Source { get; }

    /// <summary>日志正文（零拷贝引用原始字符串）。</summary>
    internal string Text { get; }
}
